src/mesh/BoundingBox.cpp

```diff
@@ -108,58 +113,68 @@ int BoundingBox::getDimension() const
 
 const std::vector<double> BoundingBox::dataVector() const
 {
-  return _bounds;
+  std::vector<double> bounds;
+  if (_dimensions == 3) {
+    bounds.insert(bounds.end(), {_boundMin[0], _boundMax[0], _boundMin[1], _boundMax[1], _boundMin[2], _boundMax[2]});
+  } else {
+    bounds.insert(bounds.end(), {_boundMin[0], _boundMax[0], _boundMin[1], _boundMax[1]});
+  }
+  return bounds;
 }
 
 void BoundingBox::expandBy(const BoundingBox &otherBB)
 {
+  PRECICE_ASSERT(_dimensions == otherBB.getDimension(), "Other BoundingBox with different dimensions than this bounding box cannot be used to expand it.");
   for (int d = 0; d < _dimensions; d++) {
-    _bounds[2 * d]     = std::min(_bounds[2 * d], otherBB._bounds[2 * d]);
-    _bounds[2 * d + 1] = std::max(_bounds[2 * d + 1], otherBB._bounds[2 * d + 1]);
+    _boundMin[d] = std::min(_boundMin[d], otherBB._boundMin[d]);
+    _boundMax[d] = std::max(_boundMax[d], otherBB._boundMax[d]);
   }
 }
 
 void BoundingBox::expandBy(const Vertex &vertices)
 {
-  PRECICE_ASSERT(_dimensions == vertices.getDimensions(), "Vertex with different dimensions than bounding box cannot be used to expand bounding box");
+  PRECICE_ASSERT(_dimensions == vertices.getDimensions(), "Vertex with different dimensions than this bounding box cannot be used to expand it.");
   const auto coords = vertices.rawCoords();
   for (int d = 0; d < _dimensions; ++d) {
-    _bounds.at(2 * d)     = std::min(coords[d], _bounds.at(2 * d));
-    _bounds.at(2 * d + 1) = std::max(coords[d], _bounds.at(2 * d + 1));
+    _boundMin[d] = std::min(coords[d], _boundMin[d]);
+    _boundMax[d] = std::max(coords[d], _boundMax[d]);
   }
 }
 
 void BoundingBox::expandBy(double value)
 {
-  if (empty())
+  // Adding or substracting a constant to the default state will still make an 'illegal' box
```

Comment:
But what is the reason this method should just return if the box is at its default state? I think what you did before with the assertion was better.

In any case, a typo:

```suggestion
  // Adding or subtracting a constant to the default state will still make an 'illegal' box
```

Reply:
Correctec